### PR TITLE
61 auto generate api documentation from json schema

### LINKED ISSUE #61: Auto generate API documentation from JSON Schema
I just discovered [doca](https://github.com/cloudflare/doca), which is a tool developed by CloudFlare, that generates HTML from JSON Schema files. Here is an introductory blog post, that shows how this works: https://blog.cloudflare.com/cloudflares-json-powered-documentation-generator/

Also: [CloudFlares own API documentation](https://api.cloudflare.com/) generated with doca

Might be worth looking into, given that we already have JSON Schema files for the API tests.

### PR DESCRIPTION
**Pull request by eugenk**
This shall fix #61. It already contains the schemas from #119 and extends them by examples and descriptions. 

The Readme.md is updated to reflect the documentation building process.

### CODE REVIEW

**Comment by eugenk** [comment]
I think we can ignore Hound's comments on the doca_config.js

File: spec/support/api/schemas/apidoc/registration.json [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+          "Content-Type": {
+            "enum": ["application/json"],
+            "example": "application/json"
+          }
+        }
+      },
+
+      "schema": {
+        "type": "object",
+        "requried": ["user"],
```
Comment (phyrog) [inline spec/support/api/schemas/apidoc/registration.json]: Typo

File: spec/support/api/schemas/apidoc/registration.json [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    {
+      "title": "Update",
+      "description": "Change the user's details. Returns HTTP 200 (OK).",
+      "rel": "self",
+      "href": "/users",
+      "method": "PATCH",
+
+      "schema": {
+        "type": "object",
+        "requried": ["user"],
```
Comment (phyrog) [inline spec/support/api/schemas/apidoc/registration.json]: Copy paste typo

**Comment by eugenk** [comment]
I rebased on master and fixed the typo.

**Pull request merged**